cli/azd/pkg/apphost/generate.go

```diff
@@ -1096,6 +1125,17 @@ func (b infraGenerator) evalBindingRef(v string, emitType inputEmitType) (string
 
 		return res, nil
 	}
+	if valueString, has := b.valueStrings[resource]; has && prop == "value" {
```

Comment:
YMMV, but with `has && prop == "value"`, it's entirely possible for `UNKNOWN_VAR_DUE_TO_BUG.value` to be left unevaluated instead of ending in error.

Reply:
`resource` is a unique name from the manifest.